source/code/plugins/npmd_config_lib.rb

```diff
@@ -304,17 +444,28 @@ def self.parse(string)
                 end
 
                 _config = _doc.elements[RootConfigTag + "/" + SolnConfigV3Tag]
+
                 if _config.nil? or _config.elements.empty?
                     Logger::logWarn "found nothing for path #{RootConfigTag}/#{SolnConfigV3Tag} in config string"
                     return nil
                 end
+                
+                @agentData = JSON.parse(_config.elements[AgentInfoTag].text())
+                @metadata = JSON.parse(_config.elements[MetadataTag].text())
 
                 _h = Hash.new
+                _h[KeyMetadata] = @metadata
                 _h[KeyNetworks] = getNetworkHashFromJson(_config.elements[NetworkInfoTag].text())
                 _h[KeySubnets]  = getSubnetHashFromJson(_config.elements[SubnetInfoTag].text())
                 _h[KeyAgents]   = getAgentHashFromJson(_config.elements[AgentInfoTag].text())
                 _h[KeyRules]    = getRuleHashFromJson(_config.elements[RuleInfoTag].text())
+                _h[KeyEpm]      = getEpmHashFromJson(_config.elements[EpmInfoTag].text())
+                _h[KeyER]       = getERHashFromJson(_config.elements[ERInfoTag].text())
+                
                 _h = nil if (_h[KeyNetworks].nil? or _h[KeySubnets].nil? or _h[KeyAgents].nil? or _h[KeyRules].nil?)
+		        if _h == nil
+		            Logger::logError "UI Config parsed as nil"
```

Comment:
Consistent whitespace as spaces instead of tabs.

Reply:
Fixed across file.